src/index.tsx

```diff
@@ -175,34 +174,25 @@ apiConnection.onMessage(msg => {
     console.warn(`Unregognized message type from server: ${type0}`)
   }
 });
-// setApiConnection(apiConnection);
-const waitForWorkspaceInfo = async () => {
-  while (true) {
-    const state = theStore.getState() as RootState
-    const workspaceInfo = state.workspaceInfo
-    if ((workspaceInfo) && (workspaceInfo.workspaceName)) {
-      apiConnection.sendMessage({
-        type: 'reportClientInfo',
-        clientInfo: {
-          feedUri: workspaceInfo.feedUri,
-          workspaceName: workspaceInfo.workspaceName,
-          readOnly: workspaceInfo.readOnly
-        }
-      })
-      return;
+
+const handleSetWorkspaceInfo = (workspaceInfo: WorkspaceInfo) => {
+  apiConnection.sendMessage({
+    type: 'reportClientInfo',
+    clientInfo: {
+      feedUri: workspaceInfo.feedUri,
+      workspaceName: workspaceInfo.workspaceName,
+      readOnly: workspaceInfo.readOnly
     }
-    await sleepMsec(10);
-  }
+  })
 }
-waitForWorkspaceInfo();
 
 const content = (
   // <React.StrictMode> // there's an annoying error when strict mode is enabled. See for example: https://github.com/styled-components/styled-components/issues/2154 
   <HitherContext.Provider value={hither}>
     <MuiThemeProvider theme={theme}>
       <Provider store={theStore}>
         <Router>
-          <AppContainer />
+          <AppContainer onSetWorkspaceInfo={handleSetWorkspaceInfo} />
```

Comment:
The AppContainer is responsible for setting the workspace info and reporting it back to this file. Perhaps this method could be improved/simplified even further.